src/Language/Spectacle/Syntax/Modal.hs

```diff
@@ -0,0 +1,165 @@
+-- | Temporal operators.
+--
+-- @since 0.1.0.0
+module Language.Spectacle.Syntax.Modal
+  ( -- * Labels
+    Modal (Modal),
+    Effect (Always, UpUntil),
+
+    -- * Syntax
+    always,
+    eventually,
+    upUntil,
+
+    -- * Interpreters
+    introducePrimes,
+    applyModality,
+  )
+where
+
+import Data.Functor.Loom (hoist, runLoom, (~>~))
+import Language.Spectacle.Lang
+  ( Lang (Op, Pure, Scoped),
+    Member (injectS, projectS),
+    Members,
+    Op (OHere, OThere),
+    Scoped (SHere, SThere),
+    scope,
+  )
+import Language.Spectacle.Syntax.Logic.Internal (Logic)
+import Language.Spectacle.Syntax.Modal.Internal (Effect (Always, UpUntil), Modal (Modal))
+import Language.Spectacle.Syntax.Plain (Effect (PlainVar))
+import Language.Spectacle.Syntax.Plain.Internal (Plain)
+import Language.Spectacle.Syntax.Prime (Effect (PrimeVar), Prime)
+
+-- ---------------------------------------------------------------------------------------------------------------------
+
+-- | The modal operator 'always' qualifies a formula @p@ such that for all actions taken in a specification, the formula
+-- @p@ must be true.
+--
+-- [Valid]
+--
+-- @
+-- ... ───── p ───── p ───── p ───── p ───── p ───── ...
+-- @
+--
+-- [Invalid]
+--
+-- @
+-- ... ───── p ───── p ───── p ───── ○ ───── p ───── ...
+-- @
+--
+-- @since 0.1.0.0
+always :: Member Modal effs => Lang ctx effs Bool -> Lang ctx effs Bool
+always m = scope (Always m)
+{-# INLINE always #-}
+
+-- | The modal operator 'eventually' qualifies a formula @p@ such that @p@ must be true /now/, or some time in the
+-- future. 'eventually' is equivalent to 'upUntil' with a trivial formula on the left-hand side:
```

Comment:
"left-hand side" is confusing here because it reads like you mean `'eventually' p` when I think what you mean is the "left-hand side of the upUntil formula: `('pure' 'True')`.